### ISSUE TITLE
Improve clarity/wording of help bubbles, add bubbles where necessity

### ISSUE THREAD

**Issue by cmiya** [cwrc/HuViz#5]
- The text should be as straightforward and direct as possible, without repeating the obvious.
- Sometimes the help bubbles are confusing and/or repetitive. For example, the help bubble for “line thickness” reads “the thickness of the line.” This is repetitive information that doesn’t clarify the meaning. Which lines? Likewise, the help bubble for “charge” under the settings tab reads “the repulsive charge between nodes.” A better choice might be: “controls how much the nodes attract and repel one another, similar to a magnetic charge.”
- Avoid spelling/grammar mistakes and informal wording. For instance, the bubble for “node radius” reads: “how fat the nodes are.” A better choice might be: “controls the size of the node.” Have also spotted minor spelling errors. For instance, the help bubble for “hide” misspells graph as “grpah.”
- Add help bubbles to clarify:
  - “Select all” button (current bubble states "All nodes are in this set, regardless of state." This would not lead me to believe that I could use this button to, for instance, label or unlabel the entire group)
  - Bold VS black node circles
  - Node colours (grey VS coloured)
  - “Like” search bar.

**Comment by cmiya** [comment]
I think this is a crucial task that needs to be completed before user testing. What might work best is setting up a shared document in GoogleDocs (perhaps in tabular form). The document could list agreed upon rules for compiling help bubbles, such as: don't just repeat the name of the object, follow standard grammar/punctuation, etc. The document could include a table with columns listing the Feature/Button (for ex: verb "choose"), Help Bubble Text, and Function explaining what the help bubble in this particular instance should do (for ex: the help bubble should answer questions like: what does the feature/button help the users accomplish? When would this feature/button be useful?) Although this will take a bit of time, I think it will be worth it to work through these kinks and establish a standard list of rules/best practices that we could use for future tools (maybe CWRC writer?) that might require help bubbles or other descriptive text.

**Comment by cmiya** [comment]
We might also think about what it is we want the help bubbles to accomplish. Under what conditions would the bubbles be useful? Who is the user we have in mind? (EX: first-time user OR repeat user in need of a refresher on the function of certain tools -> why an on/off button might be useful in the future) We could include these aims at the top of the shared GoogleDoc.

**Comment by SusanBrown** [comment]
This sounds likely to be worthwhile. I am not sure though if you mean the hover-over bubbly feature in HuViz or another kind of help bubble.

**Comment by cmiya** [comment]
Yep! Sorry if it wasn't clear (I just changed the wording of the topic title). I meant the hover-over bubbles that are currently used in the beta version.

**Comment by ilovan** [comment]
This is related to #4 .
I personally don't like the hover-over bubbles that much, since they tend to mess the accessibility of a tool and to behave poorly on touch screens. 
I also think we should run a very exigent triage on what warrants a bubble and what doesn't. (e.g. node radius is pretty self explanatory)

**Comment by cmiya** [comment]
I agree that they can be distracting and often simply restate the obvious (as in the description for line thickness). However, they might be useful if they explained under what circumstances the buttons (is this the correct term?) were useful. For instance, the Fish Eye radius helps zero in on individual nodes when dealing with large, overcrowded datasets. Sway Fraction is useful to separate the connective lines between nodes when dealing with multiple predicates.

**Comment by jsmithelford** [comment]
Perhaps it would be useful to have a button that turns on or off the bubbles or "help" function. This would be a way to avoid cluttering up the screen once the user has figured it out, but doesn't run the risk of leaving out crucial information.

**Comment by cmiya** [comment]
Jana, I totally agree. I think the help bubbles should have an on/off function. That way they can be useful, not only for new users, but more experienced users who might need a reminder of a certain tool or setting function while they are in the middle of working with a dataset.

**Comment by ilovan** [comment]
I said this before, but I am going to take the risk of repeating myself: we have to also be mindful of the hover-over functionalities in general and how they translate to touch screen environments. I can definitely see HuViz being used on tablets.

**Comment by cmiya** [comment]
I wonder how that would work in terms of the new 3D touch technology? New devices giver users a detailed glimpse of the thing if you press and hold. For example, if you are scrolling through your email and press and hold on the title of the message, it will give you a glimpse of what it contains. We could potentially utilize this for hover/help bubbles where pressing and holding would activate the popup.

**Comment by jsmithelford** [comment]
Do we need to worry about the hover-over function now, before user testing, or can we sort out the functionality in touch-screen environments after user testing?

**Comment by cmiya** [comment]
First draft of the Quick Reference Guide has been uploaded to the CWRC Documentation Folder (could use as basis for updating the Hover Bubbles):

https://docs.google.com/document/d/1L-fs8CbUQrejqva3wMbSto1ffaFpGHg8dsOg5ctVeVU/edit?usp=sharing

**Comment by smurp** [comment]
We should defer touch considerations until after user testing.  It will be a fair amount of work to properly support the various contexts, eg: screen size (phone/tablet/desktop/retina), OS (iOS/Android/etc) and browser variations. I have been mindful of touch screens by having us resist modifier-keys and context menus but haven't been taking extraordinary measures such as using the various browser-specific CSS tags pertinent of touch devices.

**Comment by cmiya** [comment]
I agree @smurp. I like the idea of the bubbles and think they'll be helpful for the test users. For now, I think we can just work on clarifying the text inside the bubbles. It should be a simple cut and paste job. I can compile the list of text (based on the quick guide notes) and pass it on to you.

**Comment by antimony27** [comment]
@cmiya Above you mention a Google Doc that outlines the wording for these bubbles. Do you know if this ever happened? I will start one if it hasn't been done.

**Comment by cmiya** [comment]
We have working hover-over help bubbles for most elements (e.g. verbs, sets, most of the settings features). I think the text just need to be tidied up a bit. Some of the newer features are lacking help bubbles (e.g. View/Edit, Current Command, Matching) - it might be useful to add them.

I created this [Reference Guide](https://docs.google.com/document/d/1L-fs8CbUQrejqva3wMbSto1ffaFpGHg8dsOg5ctVeVU/edit?usp=sharing) of HuViz terminology awhile back with the idea of adding it to the Documentation. But we could also use it as a basis for the hover-bubbles. 

We also talked about having the option to turn on/off the help bubbles (i.e. enter into "help mode"), so it's an option for new users but not distracting for experienced ones.

**Comment by antimony27** [comment]
@smurp in Alpha, the labels on the Verbs are quite finicky, and it doesn't seem like they all have them (Activate, Deactivate, Select, Unselect don't work). If @cmiya can provide access to the reference guide, perhaps we can work together on getting this in order.
